components/org.wso2.carbon.analytics.msf4j.interceptor.common/src/main/java/org/wso2/carbon/analytics/msf4j/interceptor/common/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.analytics.msf4j.interceptor.common;
+
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.analytics.idp.client.core.api.IdPClient;
+import org.wso2.carbon.analytics.idp.client.core.exception.AuthenticationException;
+import org.wso2.carbon.analytics.idp.client.core.utils.IdPClientConstants;
+import org.wso2.carbon.analytics.msf4j.interceptor.common.internal.DataHolder;
+import org.wso2.carbon.analytics.msf4j.interceptor.common.util.InterceptorConstants;
+import org.wso2.carbon.messaging.Headers;
+import org.wso2.msf4j.Request;
+import org.wso2.msf4j.Response;
+import org.wso2.msf4j.interceptor.RequestInterceptor;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Interceptor to check authentication into sp.
+ */
+@Component(
+        name = "Authentication-Interceptor",
+        service = RequestInterceptor.class
+)
+public class AuthenticationInterceptor implements RequestInterceptor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationInterceptor.class);
+
+    @Override
+    public boolean interceptRequest(Request request, Response response) throws Exception {
+        if (!DataHolder.getInstance().isInterceptorEnabled()) {
+            return true;
+        } else {
+            for (String url : DataHolder.getInstance().getExcludeURLList()) {
+                if (request.getUri().matches(url)) {
+                    return true;
+                }
+            }
+            IdPClient idPClient = DataHolder.getInstance().getIdPClient();
+            Headers headers = request.getHeaders();
+            String authorizationHeader = request.getHeader(IdPClientConstants.AUTHORIZATION_HEADER);
+            if (authorizationHeader != null) {
+                String headerPrefix = authorizationHeader.split(" ")[0];
+                String headerPostfix = authorizationHeader.split(" ")[1];
```

Comment:
What happens if the `authorizationHeader` doesn't have any spaces, then `authorizationHeader.split(" ")` will return a zero length array and accessing index `1` will result an `ArrayIndexOutOfBoundsException`

Reply:
Note that we cannot trust anything comes from the client-side

Reply:
Fixed in 344997f with a contains check